do away with sentinel errors

lestrrat opened the pull request.

natefinch commented on marshal.go:
```diff
@@ -33,12 +34,12 @@ func extractInt(n *Integer, m map[string]interface{}, s string) error {
 
 	switch v.(type) {
```
This is probably better written as 
```go
switch val := v.(type) {
case float64:
    n.Val = int(val)
    n.Initialized = true
    return nil
default:
    return errors.Wrap(errInvalidType("float64", v), "failed to extract int")
}
```
the val := v.(type) makes a new variable of the appropriate type, so you don't have to re-convert.

Or, since there's only ever one expected type, you can just do this:

```go
val, ok := v.(float64)
if !ok {
    return errors.Wrap(errInvalidType("float64", v), "failed to extract int")
}
n.Val = int(val)
n.Initialized = true
return nil
```

natefinch commented on marshal.go:
```diff
@@ -355,6 +396,47 @@ func extractDependecies(res *DependencyMap, m map[string]interface{}, name strin
+		return nil
+	}
+
+	switch v.(type) {
```
use `switch v := v.(type) {` here too

natefinch commented on marshal.go:
```diff
@@ -272,36 +299,44 @@ func extractSchemaMap(m map[string]interface{}, name string) (map[string]*Schema
+		return nil, nil
+	}
+
+	switch v.(type) {
```
use `m, ok := v.(map[string]interface{})` here, too.

lestrrat commented:
Why are you commenting on a `merged` commit?

natefinch commented:
I was looking through the latest PRs to see what's been changed, and there's no other way to produce line by line comments except on a PR, unless I just manually link to lines in an issue or something.   Seemed an easy way to give some feedback.

lestrrat commented:
Thank you for taking the time, and I really do appreciate people critiquing stuff that I write.

But the way/tone with which you are doing this makes me feel like I'm being taught by somebody who feels superior than I am. Also, this is old stuff, and the PR's motivation has nothing to do with the lines you are commenting on (well, the switch part doesn't), so it feels very very out of context. I think you are making this waaaay harder for me to swallow than it should be.

Now, that's just me being an emo. So as a realistic alternative, may I suggest that you create PRs addressing these issues? I will merge them (and, yes, point taken for each of the points you raised), and we can both get what we want?

Thanks!

natefinch commented:
Hey lestrrat,

I am really sorry about the way I handled this. I should have started a conversation before just coming in and crapping all over code from a month and a half ago.  My apologies.

I'm sorry it came across as superior.  I try to keep my review comments neutral, but I also try to explain why I'm suggesting changes.... I guess it didn't work out this time.

I'll make a PR, though honestly none of this is terribly important, they were just some minor things that stuck out when I was reading the change list.

lestrrat commented:
It would have worked perfectly if this weren't an old, merged PR! :D
And while I was a bit like "WTF?", not offended, valid points, and I emphasize that I do appreciate critiques. Waiting on your PRs!
